File: src/main/java/api/routes/locale/LocaleService.java
package api.routes.locale;

import api.utils.ApiUtils;
import api.routes.locale.dto.LocaleDto;
import lombok.NoArgsConstructor;
import main.commands.slashcommands.commands.management.LanguageCommand;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

@Service
@NoArgsConstructor
public class LocaleService {

    public ResponseEntity<LocaleResponse> setLocale(LocaleDto locale) {
        final var guild = ApiUtils.getGuild(locale.getServer_id());
        new LanguageCommand().setLocale(guild, locale.getLocale());
        return ResponseEntity.ok(LocaleResponse.builder()
                .message("You have set the locale in " + guild.getName() + " to: " + locale.getLocale())
                .build()
        );
    }
}

File: src/main/kotlin/api/routes/locale/LocaleService.kt
package api.routes.locale

import api.models.response.GenericResponse
import api.models.response.OkResponse
import api.models.service.AbstractGuildService
import main.commands.slashcommands.management.LanguageCommand
import net.dv8tion.jda.api.sharding.ShardManager

class LocaleService(shardManager: ShardManager) : AbstractGuildService(shardManager) {

    suspend fun setLocale(dto: LocaleDto): GenericResponse {
        val guild = guildUtils.getGuild(dto.server_id) ?: return noGuild(dto.server_id)
        LanguageCommand().setLocale(guild, dto.locale, shardManager)
        return OkResponse("You have set the locale in ${guild.name} to ${dto.locale.uppercase()}")
    }

}